frame/support/src/traits/hooks.rs

```diff
@@ -135,17 +138,24 @@ pub trait OnRuntimeUpgrade {
 
 	/// Execute some pre-checks prior to a runtime upgrade.
 	///
+	/// Return an encoded state digest which will be passed to `post_upgrade` after upgrading for
```

Comment:
I think we need to explain this a bit more. What is a "state digest"? More importantly, we should explain here that the byte vector is intended to contain arbitrary data that can be encoded and passed to the `post_upgrade` hook for further processing, otherwise people would not know what to do with this sudden new return value for `pre_upgrade`.

Reply:
We should just call it "state" and not "state digest"

Reply:
The comment is updated, PTAL.